client/components/global-notices/index.jsx

```diff
@@ -59,7 +70,7 @@ const NoticesList = createReactClass( {
 					duration={ notice.duration || null }
 					text={ notice.text }
 					isCompact={ notice.isCompact }
-					onDismissClick={ this.removeNotice.bind( this, notice ) }
+					onDismissClick={ this.removeNoticeStoreNotice( notice ) }
```

Comment:
We still need a bind here, no? Like `this.removeNoticeStoreNotice.bind( notice )`